transportmiddleware/transportmiddleware.go

```diff
@@ -0,0 +1,33 @@
+package transportmiddleware
+
+import (
+	"context"
+
+	"github.com/yarpc/yab/transport"
+)
+
+var registeredMiddleware TransportMiddleware
+
+// Register sets the provided transport middleware to be used on future
+// calls to Apply(). Calls to Register() will overwrite previously registered
+// middlewares; that is, only one middleware is allowed at a time.
+func Register(tm TransportMiddleware) {
```

Comment:
A little thing worth having on things that change globals like this is a way to restore the old value.

```
func Register(tm TransportMiddleware) (restore func()) {
  oldMW := registeredMiddleware
  // ..
  return func() {
    registeredMiddleware = oldMW
  }
}
```

This makes your tests cleaner because you don't have to worry about polluting global state. Tests can just do,

```
restore := Register(..)
defer restore()
```